x/README.md

```diff
@@ -0,0 +1,226 @@
+
+#### Sessions
+
+To prevent relay payment requests from scaling in size, the system aims for a concise proof. Aggregation serves as a solution, allowing each relay sent to include a signature for the cumulative compute units (CU) collected thus far, adding the CU from the latest request. Each subsequent relay further augments the claimable amount. These aggregations form sessions, where a provider requires only the last message in a session to claim rewards.
+
+To prevent providers from excessively claiming using multiple proofs from the same session, the blockchain implements double spend protection. Once a session is claimed in a specific epoch, all subsequent claims on that session are blocked. Consequently, reward claims are delayed to ensure the closure of all current epoch aggregations before claiming.
+
+#### Badges
+
+To facilitate frontend usage, which poses a risk to private keys, Lava introduces a feature called badges. These are dynamically generated private keys that receive clearance or a "badge" to operate on behalf of the subscription/developer within a specific epoch, limited to a predetermined amount of compute units (CU). These badges, along with dynamically generated signatures, are sent to providers to enable them to service the badge and subsequently claim rewards for its usage on the blockchain.
+
+### Claiming Rewards
+
+Given that providers submit claim transactions for work performed off-chain, Lava allows for delayed claiming of rewards, harmonizing with the session aggregation process described earlier. Lava implements a moving time window for claiming rewards, retaining data for pairing and claimed rewards within this temporal memory window. This approach enables providers to claim rewards deliberately without the urgency of claiming right at the epoch's end. Moreover, it accommodates situations where consumers might be unsynchronized or providers encounter network issues, transaction congestion, or downtime.
+
+Providers locally store the latest relay request, containing a cryptographic signature from the consumer developer or subscription key, as proof. When an epoch concludes, after a delay, these rewards aggregate into a transaction termed 'relay_payment,' automatically triggered by the provider service. This transaction passes verifications and accumulates compute units (CU) on behalf of the provider.
+
+Upon a subscription's termination, all provided CU (capped as per the plan policy, both total and per epoch) are factored into dividing rewards among providers. Each provider receives their portion from the subscription's cost after deducting validators' and contributors' participation. Subsequently, these rewards are divided post-commission with delegators. Consequently, provider rewards are distributed monthly per subscription, contingent upon its purchase date. Rewards for providers and their delegators can be claimed through a dualstaking transaction.
```

Comment:
not termination, but end of month.

Also, the paragraph is not clear on distributing rewards until the last sentence. It looks like the providers immediately get rewarded (by the way you put it) and only the last sentence clarifies that provider rewards are basically delegation rewards. Maybe change the phrasing a bit to emphasize that the rewards are allocated on chain and need to be claimed with a specific TX